MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```diff
@@ -2419,77 +2568,357 @@ private static void correctNonAeroFlyerBehavior(List<Entity> entityList, int boa
-                // if this entity has no guns (e.g. is a Boeing Jump Bomber)
-                if (entity.getIndividualWeaponList().isEmpty()) {
-                    loadBombs(entity, validBombChoices, campaign.getGameYear());
-                    continue;
+        boolean hasGuided = false;
+        for (int i = 0; i < bomberList.size(); i++) {
+            if (numBombers >= maxBombers) {
+                break;
+            }
+
+            Entity curBomber = bomberList.get(i);
+            boolean isUnarmed = curBomber.getIndividualWeaponList().isEmpty();
+
+            // Some fighters on ground attack may be flying air cover rather than strictly
+            // air-to-ground
+            boolean isCAP = !hasGroundTargets || (Compute.d6() == 1);
+
+            // Set minimum thrust values, with lower minimums for unarmed and ground attack,
+            // and use remaining thrust to limit hardpoints
+            if (isCAP) {
+                minThrust = isUnarmed ? 2 : ((int) Math.ceil(curBomber.getWalkMP() / 2.0));
+            } else {
+                minThrust = isUnarmed ? 2 : 3;
+            }
+            maxLoad = Math.min((int) Math.floor(curBomber.getWeight() / 5.0), (curBomber.getWalkMP() - minThrust) * 5);
+
+            // Get a random percentage (40 - 90) of the maximum bomb load for armed entities
+            if (!isUnarmed) {
+                maxLoad = (int) Math.ceil((Compute.randomInt(50) + 40) * maxLoad / 100.0);
+            }
+
+            if (maxLoad == 0) {
+                continue;
+            }
+
+            // Generate bomb load
+            generatedBombs = generateExternalOrdnance(
+                    maxLoad,
+                    isCAP,
+                    isPirate,
+                    quality,
+                    campaign.getGameYear());
+            // Whoops, go yell at the ordnance technician
+            if (Arrays.stream(generatedBombs).sum() == 0) {
+                continue;
+            }
+
+            // Set a flag if any of the ordnance guided by TAG
+            hasGuided = !hasGuided &&
+                    (generatedBombs[BombType.B_LG] > 0 ||
+                            generatedBombs[BombType.B_HOMING] > 0);
+
+
+            // Store the bomb selections as we might need to add in TAG later
+            bombsByCarrier.put(i, generatedBombs);
+
+            // Do not increment bomber count for unarmed entities
+            if (!isUnarmed) {
+                numBombers++;
+            }
+
+        }
+
+        // Load bombs onto entities. If there is guided ordnance present then randomly add some TAG
+        // pods to those without the guided ordnance
+        int tagCount = Math.min(bomberList.size(), Compute.randomInt(3));
+        for (int i = 0; i < bomberList.size(); i++) {
+            Entity curBomber = bomberList.get(i);
+
+            if (bombsByCarrier.containsKey(i)) {
+                generatedBombs = bombsByCarrier.get(i);
+
+                if (hasGuided &&
```

Comment:
Move this into a conditional above the tagCount declaration; then it can be removed from this line and line 2681, and this section can be avoided if no guided munitions have been loaded in the entire bomber fleet.

Reply:
NM, superseded by later comments.